### PR TITLE
Java port of H3 hexagonal grid library

### PR DESCRIPTION
**Pull request by iverase**
Adds a new lib project that contains the port to Java of basic functionality of the bar H3 library. The main functionality proved is:

- **geoToH3**(double lat, double lng, int res) : Given point and a resolution, it returns the h3 cell the point belongs to.
- **geoToH3Address**(double lat, double lng, int res): Given point and a resolution, it returns the h3 address the point belongs to.
-  **h3ToGeoBoundary**(long h3): Given an h3 cell, it returns the Cell boundary
-  **h3ToGeoBoundary**(String h3Address): Given an h3 address, it returns the Cell boundary
- **h3ToChildren**(long h3): Given an h3 cell, it returns the children cell
- **h3ToChildren**(String h3Address): Given an h3 address, it returns the children addresses.
- **h3ToParent**(long h3): Given an h3 cell, it returns the parent cell
- **h3ToParent**(String h3Address): Given an h3 address, it returns the parent address
- **hexRing**(long h3): Given an h3 cell, it return all neighbour cells.
- **hexRing**(String h3Address): Given an h3 address, it return all neighbour addresses.

### CODE REVIEW

File: build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/precommit/LicenseHeadersTask.java [hunk 1]
```diff
@@ -172,6 +172,8 @@ public void runRat() {
         matchers.add(subStringMatcher("GEN  ", "Generated", "ANTLR GENERATED CODE"));
         // Vendored Code
         matchers.add(subStringMatcher("VEN  ", "Vendored", "@notice"));
+        // Uber h3 port
+        matchers.add(subStringMatcher("H3   ", "Vendored", "Based on the h3 project by Uber (@uber)"));
```
Comment (iverase) [inline build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/precommit/LicenseHeadersTask.java:176]: added this to make recommit happy .

**Review by imotov — APPROVED** [review]
LGTM

**Comment by iverase** [comment]
@elasticmachine update branch

**Comment by iverase** [comment]
@elasticmachine update branch

**Pull request merged**